### DISCUSSION TITLE
anyone has a setup integration with nodemon?

### DISCUSSION THREAD

**Discussion by aelbore** [swc-project/swc#1609] [General]
currently im using ts-node for nodemon, would like to know if there any idea how i can integration swc into nodemon.

**Comment by joseDaKing** [comment]
@aelbore I found this [youtube guide](https://www.youtube.com/watch?v=7EeGHUs4zfs) about how integrate nodemon and SWC

**Reply by aelbore** [reply]
@joseDaKing i created example code for this, you don't need to compile to javascript in order nodemon will run it, you can user swc register 
https://github.com/aelbore/swc-nodemon

**Comment by Sanhajio** [comment]
You can run: `swc-node src/app.ts`

or 

In your `package.json`
```
{
  "scripts": {
    "start": "swc-node src/app.ts",
    "dev": "nodemon --exec 'swc-node' src/app.ts"
  },  
}
```

**Comment by VladimirEdelmann** [comment]
If you're looking for a faster alternative to TypeScript's default compiler, you might want to consider using the swc compiler instead. With swc, you can transpile your TypeScript code into JavaScript four times quickly than with the default compiler.

1. Install SWC and SWC-Node as dev dependencies:

```
npm i -D swc swc-node
```

2. Add the following script to your package.json file:

```
"scripts": {
  "dev": "nodemon --exec node -r @swc-node/register main.ts"
}
```
3. To start your development server with nodemon and swc-node, run the following command in your project directory:

```
npm run dev
```
